specification/servicefabric/data-plane/Microsoft.ServiceFabric/stable/9.0/servicefabric.json

```diff
@@ -28048,7 +28226,6 @@
       "description": "The version of the API. This parameter is required and its value must be '8.2'.\n\nService Fabric REST API version is based on the runtime version in which the API was introduced or was changed. Service Fabric runtime supports more than one version of the API. This version is the latest supported version of the API. If a lower API version is passed, the returned response may be different from the one documented in this specification.\n\nAdditionally the runtime accepts any version that is higher than the latest supported version up to the current version of the runtime. So if the latest API version is 6.0 and the runtime is 6.1, the runtime will accept version 6.1 for that API. However the behavior of the API will be as per the documented 6.0 version.",
       "x-ms-enum": {
```

Comment:
It seems that, if this definition is for 9.0, then this enum for 8.2 should be removed.

Reply:
so, the way SF API versioning works is that the API version is pinned to when it was last changed. So our API can be fairly specific for the expected API version. For backward compatibility, we allow lower API versions to be passed in but put a constraint on the highest version that can be submitted, so with a 9.0 release we will have APIs that take anywhere say 6.0 to 9.0 as the valid version. This versioning is specific to how SF is structured and has to be maintained this way for backwards compatibility. This design has been approved by the API board.